org.eclipse.xtext.purexbase/src-gen/org/eclipse/xtext/purexbase/services/PureXbaseGrammarAccess.java

```diff
@@ -994,8 +1000,8 @@ public ParserRule getXListLiteralRule() {
 	//XClosure XExpression:
 	//	=> ({XClosure}
 	//	'[')
-	//	=> ((declaredFormalParameters+=JvmFormalParameter (',' declaredFormalParameters+=JvmFormalParameter)*)?
-	//	explicitSyntax?='|')?
+	//	=> ((declaredFormalParameters+=JvmFormalParameter (',' declaredFormalParameters+=JvmFormalParameter)*)
+	//	? explicitSyntax?='|')?
```

Comment:
Unexpected line-break before the cardinality.

Reply:
This is because of the autowrap in the following code:
```
	private void formatParens(EObject element, IFormattableDocument doc) {
		doc.prepend(regionFor(element).keyword("("), it -> it.autowrap());
		doc.append(regionFor(element).keyword("("), it -> it.noSpace());
		doc.prepend(regionFor(element).keyword(")"), it -> it.noSpace());
		doc.append(regionFor(element).keyword(")"), it -> it.autowrap());
	}
```
We allow autowrap after paranthesis ... guess we want to have that, to break lines at reasonable places ... not sure how to make it better ...

Reply:
Have it ... allow autowrap after the question mark as well:
```
	private void formatCardinality(EObject element, IFormattableDocument doc) {
		regionFor(element).keywords("?", "*", "+").forEach(r -> doc.prepend(r, it -> it.noSpace()));
		regionFor(element).keywords("?", "*", "+").forEach(r -> doc.append(r, it -> it.autowrap()));
	}

```